docs/model_developers/unmanaged_resources/shared_attributes_example_handler.py

```diff
@@ -0,0 +1,121 @@
+import re
+from collections import abc
+from typing import Optional
+
+import pydantic
+
+from inmanta import resources
+from inmanta.data.model import ResourceIdStr
+from inmanta.agent.handler import provider, DiscoveryHandler, HandlerContext, CRUDHandler
+from inmanta.resources import resource, DiscoveryResource, PurgeableResource
+
+
+class Credentials:
+    fields = ("host", "username", "password")
+
+    host: str
+    username: str
+    password: str
+
+    @staticmethod
+    def get_host(exporter, resource):
+        return resource.host.name
+
+
+@resource("my_module::Interface", agent="host.name", id_attribute="name")
+class Interface(Credentials, PurgeableResource):
+    fields = ("name", "ip_address")
+
+    name: str
+    ip_address: str
+
+
+@resource("my_module::InterfaceDiscovery", agent="host.name", id_attribute="host")
+class InterfaceDiscovery(Credentials, DiscoveryResource):
+    fields = ("name_filter",)
+
+    name_filter: Optional[str]
+
+
+class UnmanagedInterface(pydantic.BaseModel):
+    """
+    Datastructure used by the InterfaceDiscoveryHandler to return the attributes
+    of its discovered resources.
+    """
+    host: str
+    interface_name: str
+    ip_address: pydantic.IPvAnyAddress
+
+
+class Authenticator:
+    """
+    Helper class that handles the authentication to the remote host.
+    """
+
+    def login(self, credentials: Credentials) -> None:
+        raise NotImplementedError()
+
+    def logout(self, credentials: Credentials) -> None:
+        raise NotImplementedError()
+
+
+@provider("my_module::Interface", name="interface_handler")
+class InterfaceHandler(Authenticator, CRUDHandler[Interface]):
+    """
+    Handler for the interfaces managed by the orchestrator.
+    """
+
+    def pre(self, ctx: HandlerContext, resource: Interface) -> None:
+        self.login(resource)
```

Comment:
Did you run mypy on this? Again, I think it should work, but best to double check.